### PR TITLE
Adding missing Solarian Revelations

### PR DESCRIPTION
**Pull request by rmorgens**
Add missing Solarian Stellar/Zenith Revelations and update the links in the index features

### CODE REVIEW

**Comment by LebombJames** [comment]
Before you get to in-depth with this, #972 implements a helper to include graviton and photon icons without ProseMirror throwing a fit. Best to wait for that to be merged

**Comment by LebombJames** [comment]
Ok, #972 is merged. See that PR for a guide on how to use the helper.

**Comment by rmorgens** [comment]
Addresses Issue#933 https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/issues/933

File: src/items/class-features/attractive_force_(su).json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      "value": null
+    },
+    "requirements": "6th Level",
+    "rollNotes": "",
+    "save": {
+      "type": "",
+      "dc": null,
+      "descriptor": ""
+    },
+    "source": "Armory, pg. 152",
```
Comment (LebombJames) [inline src/items/class-features/attractive_force_(su).json:104]: The source for Armory is AR. See contributing.md

File: src/items/class-features/attractive_force_(su).json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:graviton]Graviton Revelation</p>\n<p>Your weapon exerts a pull that makes objects cling to it and can even stymie opponents with strands of force. As a move action, you can grant a weapon you wield, including your @UUID[Compendium.sfrpg.class-features.WWHUAfgpvk6uIgY0]{Solar Weapon}, the <strong>disarm</strong> weapon special property. This benefit lasts for 1 round or until you leave graviton mode.</p>\n<p>When you are attuned or fully attuned, if you score a critical hit with the weapon, your target must succeed at a Reflex save or be @UUID[Compendium.sfrpg.conditions.nFBcsp5FS1Wi46Fc]{Entangled} for 1 round. If the weapon already has a critical hit effect, when you score a critical hit, you can apply either the weapon’s normal critical hit effect or the entangling effect.</p>"
```
Comment (LebombJames) [inline src/items/class-features/attractive_force_(su).json:40]: `disarm` doesn't need to be bold

File: src/items/class-features/dark_matter_(su).json [hunk 3]
```diff
@@ -39,7 +39,44 @@
       "unidentified": "",
       "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub><img style=\"border: 1px solid var(--color-border-dark);\" src=\"systems/sfrpg/images/cup/gameplay/graviton.webp\" alt=\"\" width=\"20\" height=\"20\">&nbsp;</sub>Graviton Revelation<span></span></p>\n<p>As a move action, you can draw on the properties of dark matter to increase your density, allowing you to resist physical damage. You gain damage reduction 1/—. This increases to DR 2/— at 6th level and increases by 1 again every 3 solarian levels thereafter. This benefit lasts for 1 round or until you leave graviton mode. When you are attuned or fully attuned, your DR from dark matter is equal to half your solarian level.</p>\n<p><em>Currently, the modifier in Modifiers tab will need be to be opened then saved to update Damage Reduction. This required whenever activating/deactivating this feature, or attuning/unattuning.</em></p>"
```
Comment (LebombJames) [inline src/items/class-features/dark_matter_(su).json:40]: Remove the note from the end. That is no longer true.

File: src/items/class-features/attractive_force_(su).json (hunk 2 above)
Comment (LebombJames) [inline src/items/class-features/attractive_force_(su).json:40]: Clean up the empty `<sub>` element before the Graviton icon, and add a space between the icon and the label. Same is true on other items

File: src/items/class-features/burn_enchantment_(su).json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:photon]Photon Revelation</p>\n<p>Your attacks’ cosmic energy shreds the magical forces wielded by spellcasters. As a move action, you sheathe your body in protective energy that attunes to spellcasters you strike. If the next successful melee attack you make before the beginning of your next turn hits a creature capable of casting spells or using spell-like abilities, you gain a +2 enhancement bonus to saving throws against that creature’s spells and spell-like abilities for 1 round.</p>\n<p>When you are attuned or fully attuned, your weapon gains the dispelling critical effect. If the weapon already has a critical hit effect, when you score a critical hit, you can apply either the weapon’s normal critical hit effect or the dispelling effect. A target affected by the dispelling critical hit effect is the subject of a targeted @UUID[Compendium.sfrpg.spells.KQ4vSfG9tq8EJxir]{Dispel Magic}, per the spell, using your solarian level as the caster level.</p>"
```
Comment (LebombJames) [inline src/items/class-features/burn_enchantment_(su).json:57]: Spells should be italicized. Italicize the link to dispel magic. Same applies to other spells

File: src/items/class-features/burn_enchantment_(su).json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        "_id": "435883cc-fbcf-4810-8c02-0bc2a31e51a4",
+        "name": "Burn Enchantment (Su)",
+        "type": "enhancement",
+        "condition": "",
+        "effectType": "spell-save-dc",
+        "enabled": true,
+        "max": 2,
+        "modifier": "2",
+        "modifierType": "constant",
+        "notes": "To saving throws against the hit creature’s spells and spell-like abilities",
```
Comment (LebombJames) [inline src/items/class-features/burn_enchantment_(su).json:114]: You can include the "You gain a +2 enhancement bonus" in this so we're not unnecessarily cutting off text.

File: src/items/class-features/celestial_aegis_(su).json [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:photon]Photon Revelation</p>\n<p>As a move action, you channel the Positive Energy Plane’s power directly into your @UUID[Compendium.sfrpg.class-features.zM7oc1qiNLAPoH1J]{Solar Armor} or @UUID[Compendium.sfrpg.class-features.ylRInRfOfx6nsbUv]{Solar Shield}, helping you capture and redirect your escaping life energy to bolster your body. As a reaction when you take damage that causes you to lose at least 2 Hit Points, you gain temporary Hit Points equal to half the number of Hit Points you lost from the attack or effect. These temporary Hit Points last 3 rounds.</p><p class=\"MsoNormal\">If you’re attuned or fully attuned, you also gain spell resistance equal to 11 + your solarian level against necromancy spells, and against spells with the cold, darkness, or death descriptors.</p>"
```
Comment (LebombJames) [inline src/items/class-features/celestial_aegis_(su).json:40]: Get rid of `class=\"MsoNormal\"`. How'd that get there? Appears in a few other places

File: src/items/class-features/celestial_aegis_(su).json [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+      "value": null
+    },
+    "requirements": "14th Level",
+    "rollNotes": "",
+    "save": {
+      "type": "fort",
+      "dc": "10+floor(@classes.solarian.levels/2) + @abilities.cha.mod",
+      "descriptor": "negate"
+    },
+    "source": "Drift Crisis, pg. 45",
```
Comment (LebombJames) [inline src/items/class-features/celestial_aegis_(su).json:104]: Drift Crisis's Source is `DC`

File: src/items/class-features/celestial_clock_(su).json [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+      "value": null
+    },
+    "requirements": "2nd Level",
+    "rollNotes": "",
+    "save": {
+      "type": "fort",
+      "dc": "10+floor(@classes.solarian.levels/2) + @abilities.cha.mod",
+      "descriptor": "negate"
+    },
+    "source": "HotV#2 (AP#41), pg 52",
```
Comment (LebombJames) [inline src/items/class-features/celestial_clock_(su).json:137]: Just `AP #41`

File: src/items/class-features/dark_interaction_(su).json [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:graviton]Graviton Revelation</p>\n<p>By transforming your own body into dark matter, you briefly cease to interact with physical objects. Once per day when you are attuned or fully attuned, you can activate this power as a swift action to become <strong>incorporeal</strong> until the beginning of your next turn.</p>"
```
Comment (LebombJames) [inline src/items/class-features/dark_interaction_(su).json:40]: Incoporeal doesn't need to be strong

File: src/items/class-features/dark_matter_senses_(su).json [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p>@Icon[type:graviton]<sub>&nbsp;</sub>Graviton Revelation</p>\n<p>Your sensitivity to dark matter allows you to navigate in lightless conditions. You gain @UUID[Compendium.sfrpg.racial-features.HhrIJeHTxSGVX9mV]{Darkvision} with a range of 60 feet.</p>"
```
Comment (LebombJames) [inline src/items/class-features/dark_matter_senses_(su).json:40]: Don't link to  a specific species' species feature. Link to the journal describing darkvision

File: src/items/class-features/particle_field_(su).json [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    },
+    "duration": {
+      "units": "",
+      "value": ""
+    },
+    "formula": "",
+    "isActive": null,
+    "modifiers": [
+      {
+        "_id": "f5853302-84db-4a46-9753-2ca7337ff897",
```
Comment (LebombJames) [inline src/items/class-features/particle_field_(su).json:88]: Get rid of this modifier, this is not a way to grant Temp HP

File: src/items/class-features/particle_wave_(su).json [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+      ]
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.UVNOjH2ERlQjcGnU]{Zenith Revelation}</p>\n<p><span>&nbsp;</span>@Icon[type:photon]Photon Revelation</p>\n<p>When you’re fully photon-attuned, you can dissolve into a Large cloud composed of thousands of flaming pinpoints of brilliance as a standard action. While in this form, you gain a swarm attack that deals 4d6 fire damage, swarm defenses, swarm immunities (Starfinder Alien Archive 157), and a fly speed equal to your land speed. You lose all normal modes of attack and movement, all abilities that depend on your physical form, and any bonus to your KAC and EAC granted by your armor. You can maintain this form for a number of rounds equal to your solarian level. Returning to your normal form before the duration expires requires a standard action. The damage of your swarm attack increases by 1d6 for every 3 solarian levels you have beyond 7th. At 17th level, targets damaged by your swarm attack must succeed at a Reflex saving throw or be blinded for 1 round.</p>"
```
Comment (LebombJames) [inline src/items/class-features/particle_wave_(su).json:57]: Link to blinded

File: src/items/class-features/stellar_revelation.json [hunk 13]
```diff
@@ -37,9 +37,46 @@
...
...ty (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.68A5h7M3zfvlQieD]{Orbital Awareness (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.px7wLH2eAAyf3cPl]{Particle Field (Su)}</p>\n<p>@Icon[type:graviton]@UUID[Compendium.sfrpg.class-features.rAyI4nvDB8NAeCx6]{Solar Fortification (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.4EWgS85b5oEUJBxv]{Sunbolt (Su)}</p>\n<p>@Icon[type:graviton]@UUID[Compendium.sfrpg.class-features.bLFSPxAj9LPyoVAf]{Tenebrous Bulwark (Su)}</p><h2>16th Level</h2><p>@Icon[type:graviton]@UUID[Compendium.sfrpg.class-features.E3yS1oVNjLlo2A6A]{Adaptive Graviton (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.DOizxY4nw33L8Ld3]{Adaptive Photon (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.dmuliDXdehEHv1Mv]{Superlative Constellation (Su)}</p>\n<p>@Icon[type:graviton]@UUID[Compendium.sfrpg.class-features.OLXmy2N2DRCNDbDU]{Superlative Nebula (Su)}</p>\n<p>@Icon[type:graviton]@UUID[Compendium.sfrpg.class-features.MwQV4bUQ4OAWv4Yr]{Ultimate Graviton (Su)}</p>\n<p>@Icon[type:photon]@UUID[Compendium.sfrpg.class-features.SDInkeo4eWICc5cm]{Ultimate Photon (Su)}</p>"
```
Comment (LebombJames) [inline src/items/class-features/stellar_revelation.json:40]: Add spaces between the icons and the links

File: src/items/class-features/subduing_beams_(su).json [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:photon]Photon Revelation</p>\n<p>While the sun can sear and burn, it also represents a gentler force of growth and redemption. As a move action, you can grant a weapon you wield, including your @UUID[Compendium.sfrpg.class-features.WWHUAfgpvk6uIgY0]{Solar Weapon}, the <strong>nonlethal</strong> weapon special property by sheathing it in a field of cushioning energy. This lasts for 1 round or until you leave photon mode.</p>\n<p>When you are attuned or fully attuned, your weapon gains the @UUID[Compendium.sfrpg.rules.O5GDQUKPCJ9Zbnfj.JournalEntryPage.zOJ86MDq6kNzAAbB#staggered]{Staggered} critical hit effect. If the weapon already has a critical hit effect, when you score a critical hit, you can apply either the weapon’s normal critical hit effect or the staggered effect.</p>"
```
Comment (LebombJames) [inline src/items/class-features/subduing_beams_(su).json:40]: Nonlethal doesn't need to be strong, link to a journal if necessary

File: src/items/class-features/tenebrous_bulwark_(su).json [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
...Stellar Revelation}</p>\n<p>@Icon[type:graviton]Graviton Revelation</p>\n<p>As a move action, you channel the Negative Energy Plane’s power directly into your @UUID[Compendium.sfrpg.class-features.kCvI3xvm0Q1slajx]{Solar Flare} or @UUID[Compendium.sfrpg.class-features.WWHUAfgpvk6uIgY0]{Solar Weapon}, helping you capture and convert your escaping life energy into deadly force. As a reaction when you take damage that causes you to lose at least 2 Hit Points, the next successful attack you make with your @UUID[Compendium.sfrpg.class-features.kCvI3xvm0Q1slajx]{Solar Flare} or @UUID[Compendium.sfrpg.class-features.WWHUAfgpvk6uIgY0]{Solar Weapon} before the end of your next turn deals additional cold damage equal to half the number of Hit Points you lost. Undead creatures are immune to this additional cold damage.</p>\n<p>If you’re attuned or fully attuned, whenever you use this revelation’s reaction to add cold damage to your next attack, your @UUID[Compendium.sfrpg.class-features.kCvI3xvm0Q1slajx]{Solar Flare} or @UUID[Compendium.sfrpg.class-features.WWHUAfgpvk6uIgY0]{Solar Weapon} also gains the <strong>penetrating</strong> weapon special property until the end of your next turn.</p>"
```
Comment (LebombJames) [inline src/items/class-features/tenebrous_bulwark_(su).json:40]: Penetrating doesn't need to be strong

File: src/items/class-features/vivacious_nimbus_(su).json [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>@Icon[type:photon]Photon Revelation</p>\n<p>As a standard action while shedding normal or bright light with your @UUID[Compendium.sfrpg.class-features.7VVFPck5IJhTmP7q]{Flare (Su)} revelation, you can infuse the light with the Positive Energy Plane’s life-giving influence, granting creatures in the illuminated area the effects of the @UUID[Compendium.sfrpg.spells.QtZHSuJf0hcryYtR]{Death Ward} spell. This additional effect lasts for 1 round or until you leave photon mode, whichever happens first, though you can maintain this effect as a standard action each round (or as a move action if you’re fully attuned). When you @UUID[Compendium.sfrpg.conditions.tOpF6qjnlnKc10t7]{Blind} an undead creature with your @UUID[Compendium.sfrpg.class-features.7VVFPck5IJhTmP7q]{Flare (Su)} revelation flash of brilliant light, the target also takes <strong>force damage</strong> equal to your solarian level.</p>\n<p>You must have the @UUID[Compendium.sfrpg.class-features.7VVFPck5IJhTmP7q]{Flare (Su)} revelation to select this revelation.</p>"
```
Comment (LebombJames) [inline src/items/class-features/vivacious_nimbus_(su).json:40]: Force damage doesn't need to be strong

**Review by LebombJames — CHANGES_REQUESTED** [review]

File: src/items/class-features/dark_matter_(su).json (hunk 3 above)
Comment (rmorgens) [reply src/items/class-features/dark_matter_(su).json:40]: Didn't touch that, but I can fix that with this update.

File: src/items/class-features/dark_matter_senses_(su).json (hunk 10 above)
Comment (rmorgens) [reply src/items/class-features/dark_matter_senses_(su).json:40]: Can you provide some pointers on how to find the correct journal?

File: src/items/class-features/dark_matter_(su).json (hunk 3 above)
Comment (LebombJames) [reply src/items/class-features/dark_matter_(su).json:40]: I know yeah, just asking you to axe it while we're working on it

File: src/items/class-features/dark_matter_senses_(su).json (hunk 10 above)
Comment (LebombJames) [reply src/items/class-features/dark_matter_senses_(su).json:40]: Off the top of my head, I believe Darkvision is under the ability descriptions journal in the SFRPG rules compendium. Or maybe in the senses journal if that exists?

File: src/items/class-features/dark_matter_(su).json [hunk 17]
```diff
@@ -37,7 +37,7 @@
...
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.G7p2YIGFTAU6uGjL]{Stellar Revelation}</p>\n<p><sub>&nbsp;</sub>Graviton Revelation</p>\n<p>As a move action, you can draw on the properties of dark matter to increase your density, allowing you to resist physical damage. You gain damage reduction 1/—. This increases to DR 2/— at 6th level and increases by 1 again every 3 solarian levels thereafter. This benefit lasts for 1 round or until you leave graviton mode. When you are attuned or fully attuned, your DR from dark matter is equal to half your solarian level.</p>"
```
Comment (LebombJames) [inline src/items/class-features/dark_matter_(su).json:40]: Don't think you added an `@Icon` after removing the old image? Still got an empty `sub` too

**Review by LebombJames — CHANGES_REQUESTED** [review]

**Comment by rmorgens** [comment]
I think that has resolved all the identified issues

**Pull request merged**